### PR TITLE
Add dynamic placement changing for popover

### PR DESCRIPTION
**Pull request by ousram**

### CODE REVIEW

**Comment by sorich87** [comment]
I think we need to add another option to trigger that behaviour (it can be true by default).

File: src/coffee/bootstrap-tour.coffee [hunk 1]
```diff
@@ -473,9 +473,34 @@
...
+        top:$(window).scrollTop()
+        bottom:$(window).scrollTop() + $(window).height()
+        left:$(window).scrollLeft()
+        right:$(window).scrollLeft() + $(window).width()
+      }
+
       $element
       .popover(
-        placement: step.placement
+        placement: (popover,element)->
```
Comment (sorich87) [inline src/coffee/bootstrap-tour.coffee]: What about putting this logic out of _showPopover?

Comment (ousram) [inline src/coffee/bootstrap-tour.coffee]: I wonder if that logic will be useful for another purpose. If not, I guess it's ok here, or do you suggest us to create another function for this ?

**Comment by ousram** [comment]
Like smart-placement, or something like that ?

File: src/coffee/bootstrap-tour.coffee (hunk 1 above)
Comment (sorich87) [inline src/coffee/bootstrap-tour.coffee]: Yes, so much code there is a bit difficult to read

**Comment by sorich87** [comment]
Yes, smartPlacement would be good

**Comment by ousram** [comment]
I'm about to submit a new PR.

**Pull request closed without merging**